src/StoreApi/Utilities/CartController.php

```diff
@@ -471,22 +470,26 @@ public function get_shipping_packages( $calculate_rates = true ) {
 			return [];
 		}
 
-		$packages = $cart->get_shipping_packages();
+		$packages = apply_filters( '__experimental_woocommerce_store_api_cart_shipping_packages', $cart->get_shipping_packages() );
```

Comment:
I'm concerned about how this increases the surface area of how extensions could break the shipping data (i.e. remove existing packages etc). If this approach is taken, ideally we'd surface via an explicit registry that extensions would register callbacks to for enhancing the packages array and we'd merge any packages returned from callbacks in the registry when assembling this data.

Reply:
Yes that's going to be the way to go if we want to continue with this approch.

Reply:
This is why I was pushing for this to be handled by subscriptions under a different key - adding packages in this context outside of the context of the actual cart doesn't make total sense to me.